File: web/src/main/java/by/sorface/passport/web/services/tokens/TokenValidator.java
package by.sorface.passport.web.services.tokens;

import by.sorface.passport.web.dao.models.TokenEntity;
import by.sorface.passport.web.dao.models.enums.TokenOperationType;

public interface TokenValidator {

    void validateOperation(final TokenEntity token, final TokenOperationType operationType);

    void validateExpiredDate(final TokenEntity token);

}

File: web/src/main/kotlin/by/sorface/passport/web/services/tokens/TokenValidator.kt
package by.sorface.passport.web.services.tokens

import by.sorface.passport.web.dao.models.TokenEntity
import by.sorface.passport.web.dao.models.enums.TokenOperationType

interface TokenValidator {
    fun validateOperation(token: TokenEntity?, operationType: TokenOperationType)

    fun validateExpiredDate(token: TokenEntity)
}
